File: 14-project-euler/Python/1-fizzbuzz.py
import sys
from typing import Mapping


def fizzbuzz(n: int, rules: Mapping[int,str]) -> list[str]:
    ans: list[str] = []
    for i in range(1,n+1):
        aux: str = ''
        for idx, elem in rules.items():
            if i%idx == 0:
                aux+=elem
        ans.append(aux or str(i))
    return ans


def main():
    n = int(sys.argv[1])
    lower = int(sys.argv[2])
    upper = int(sys.argv[3])
    rules = {lower: "Fizz", upper: "Buzz"}
    print(*fizzbuzz(n, rules), sep='\n')

if __name__ == '__main__':
    main()

File: 14-project-euler/Rust/1-fizzbuzz.rs
fn fizzbuzz(n: i32, lower: i32, upper: i32) -> Vec<String> {
    let mut ans = Vec::new();
    for i in 1..=n {
        let mut aux = String::new();
        if i%lower == 0 {
            aux.push_str("Fizz");
        }
        if i%upper == 0 {
            aux.push_str("Buzz");
        }
        if i%lower!=0 && i%upper!=0 {
            aux += &i.to_string();
        }
        ans.push(aux);
    }
    ans
}

fn main(){
    println!("{0:?}", fizzbuzz(15, 3, 5));
}